heartbeat/monitors/active/http/respbody.go

```diff
@@ -20,69 +20,79 @@ package http
 import (
 	"crypto/sha256"
 	"encoding/hex"
-	"fmt"
 	"io"
 	"net/http"
 	"unicode/utf8"
 
-	"github.com/elastic/beats/heartbeat/eventext"
-	"github.com/elastic/beats/libbeat/common"
-
 	"github.com/elastic/beats/heartbeat/reason"
-	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
 )
 
-func handleRespBody(event *beat.Event, resp *http.Response, responseConfig responseConfig, errReason reason.Reason) error {
-	defer resp.Body.Close()
-
-	sampleMaxBytes := responseConfig.IncludeBodyMaxBytes
-
-	includeSample := responseConfig.IncludeBody == "always" || (responseConfig.IncludeBody == "on_error" && errReason != nil)
-
-	// No need to return any actual body bytes if we'll discard them anyway. This should save on allocation
-	if !includeSample {
-		sampleMaxBytes = 0
+// maxBufferBodyBytes sets a hard limit on how much we're willing to buffer for any reason internally.
+// since we must buffer the whole body for body validators this is effectively a cap on that.
+// 100MiB out to be enough for everybody.
+const maxBufferBodyBytes = 100 * 1024 * 1024
```

Comment:
Would probably make sense to document this max value somewhere in the docs. At least 1 user will hit it :-)

Reply:
++